plugins/workflow/workflow_pool.go

```diff
@@ -27,7 +37,7 @@ type workflowPool struct {
 }
 
 // NewWorkflowPool creates new workflow pool.
-func NewWorkflowPool(ctx context.Context, factory server.Server) (*workflowPool, error) {
+func NewWorkflowPool(ctx context.Context, listener util.EventListener, factory server.Server) (*workflowPool, error) {
```

Comment:
```suggestion
func NewWorkflowPool(ctx context.Context, listener util.EventListener, factory server.Server) (WorkflowPool, error) {
```
Return public interface instead of a pointer to the private structure.

Reply:
This is private structure, I would like to keep it private. I'll change to method to be private as well.

Reply:
> This is private structure, I would like to keep it private. I'll change to method to be private as well.

You may create a private interface